kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -27,59 +28,59 @@ const ContentNodeResource = coreApp.resources.ContentNodeResource;
+  const summaryPayload = {
+    channel_id: channel.id,
+    collection_kind: ReportConstants.UserScopes.CLASSROOM,
+    collection_id: classId,
+  };
 
-function _showChannelList(store, classId) {
+  // workaround for conditionalPromise.then() misbehaving
+  return new Promise(
+    (resolve, reject) => {
+      const getSumm = ContentSummaryResource.getModel(channel.root_id, summaryPayload).fetch();
+      getSumm.then(
+        channelSummary => {
+          resolve({ [channel.id]: channelSummary.last_active });
+        },
+        error => reject(error)
+      );
+    }
+  );
+}
+
+function getAllChannelsLastActivePromise(channels, classId) {
+  const promises = channels.map((channel) => channelLastActivePromise(channel, classId));
+  return Promise.all(promises);
+}
+
+function _showChannelList(store, classId, showRecentOnly = false) {
   // don't handle super users
   if (coreGetters.isSuperuser(store.state)) {
     store.dispatch('SET_PAGE_STATE', {});
     store.dispatch('CORE_SET_PAGE_LOADING', false);
     store.dispatch('CORE_SET_ERROR', null);
-    return;
+    return Promise.resolve();
   }
 
-  function channelLastActivePromise(channel) {
-    // helper function for _showChannelList
-    // @param channel to get recentActivity for
-    // @returns promise that resolves channel with lastActive value in object:
-    // {
-    //   'channelId': dateOfLastActivity,
-    // }
-    const summaryPayload = {
-      channel_id: channel.id,
-      collection_kind: ReportConstants.UserScopes.CLASSROOM,
-      collection_id: classId,
-    };
-
-    // workaround for conditionalPromise.then() misbehaving
-    return new Promise(
-      (resolve, reject) => {
-        const getSumm = ContentSummaryResource.getModel(channel.root_id, summaryPayload).fetch();
-        getSumm.then(
-          channelSummary => {
-            const channelLastActive = {};
-            channelLastActive[channel.id] = channelSummary.last_active;
-            resolve(channelLastActive);
-          },
-          error => reject(error)
-        );
-      }
-    );
-  }
-
-  const channelLastActivePromises = [];
-  store.state.core.channels.list.forEach(
-    channel => channelLastActivePromises.push(channelLastActivePromise(channel))
-  );
-  channelLastActivePromises.push(setClassState(store, classId));
+  const promises = [
+    getAllChannelsLastActivePromise(store.state.core.channels.list, classId),
+    setClassState(store, classId),
+  ];
 
-  Promise.all(channelLastActivePromises).then(
-    allChannelLastActive => {
-      const lastActive = {};
-      allChannelLastActive.forEach(
-        channelLastActive => Object.assign(lastActive, channelLastActive)
-      );
-      const pageState = { lastActive };
-      store.dispatch('SET_PAGE_STATE', pageState);
+  return Promise.all(promises).then(
+    ([allChannelLastActive]) => {
+      store.dispatch('SET_PAGE_STATE', {
+        lastActive: assign({}, ...allChannelLastActive),
```

Comment:
What's the motivation for using lodash assign here, rather than Object.assign? (note that we already polyfill Object.assign)